tests/org.jboss.tools.central.ui.bot.test/src/org/jboss/tools/central/test/ui/reddeer/HTML5Parameterized.java

```diff
@@ -304,4 +351,36 @@ private void checkDeployedProject(String projectName) {
 		return projects;
 	}
 
+	private static boolean isProjectMultimodular(String exampleName) {
+		String[] multimoduleProject = { "app-client", "ejb-throws-exception", "kitchensink-ear", "kitchensink-ml-ear" };
+		return ArrayUtils.contains(multimoduleProject, exampleName);
+	}
+	
+	private static void checkDeployedSubmodules(String topNodeProjectName, List<Project> leafNodeProjects) {
+		ServersView2 serversView = new ServersView2();
+		serversView.open();
+		Server server = serversView.getServer(FULL_SERVER_NAME);
+
+		TreeItem serverTreeItem = server.getTreeItem();
+		List<TreeItem> serverChildrenTreeItems = serverTreeItem.getItems();
+		List<TreeItem> deployedSubmodulesTreeItems = new ArrayList<TreeItem>();
```

Comment:
You can use https://github.com/eclipse/reddeer/blob/37f27affd709a9622d84e8d29d047dc78a0de1b4/plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/wst/server/ui/cnf/Server.java#L215 to simplify code

Reply:
Well, the "getModules" method does not return the modules, it returns only the top TreeItem in the Servers view (**TreeItem: Red Hat JBoss Enterprise Application Platform 7.0 Server  [Started, Synchronized] > jboss-app-client-ear(jboss-app-client)  [Started, Synchronized]**):
 
<img width="877" alt="The Servers View" src="https://user-images.githubusercontent.com/43820055/61213474-6c96ad80-a705-11e9-99f7-a3d41abc5270.png">

Reply:
Here is the "Variables" view and the serverModules variable, which content is created using the getModules function (**List<ServerModule> serverModules = server.getModules();**):

<img width="786" alt="The Variables View" src="https://user-images.githubusercontent.com/43820055/61213579-b8e1ed80-a705-11e9-98c6-b70b6ceaf5b9.png">

Reply:
well, you are right, that functions for sub modules are not implemented, but function getModules() returns correct modules. EAR is an archive which contains client side and server side and dependencies, when you try different workflow(add/remove modules from server) you will see only app-client-ear, not submodules(client-simple, ejb).

Reply:
@zcervink Please create issue in RedDeer for missing functionality.

Reply:
@jkopriva issue created - https://issues.jboss.org/browse/JBIDE-26742